python3Packages.django-lasuite: init at 0.0.9

soyouzpanda opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

soyouzpanda committed:
python3Packages.django-lasuite: init at 0.0.9

ethancedwards8 commented on pkgs/development/python-modules/django-lasuite/default.nix:
```diff
@@ -0,0 +1,73 @@
+  pythonImportsCheck = [
+    "lasuite"
+  ];
+
```
consider adding pytestCheckHook

ethancedwards8 reviewed (changes requested):
otherwise LGTM.

mweinelt commented on pkgs/development/python-modules/django-lasuite/default.nix:
```diff
@@ -0,0 +1,83 @@
+    hash = "sha256-0UeXMqpk7DPWAGZto7nHhvFK6YsNqywrTFQ/q+7IHDY=";
+  };
+
+  preCheck = ''
+    export PYTHONPATH=tests:$PYTHONPATH
+    export DJANGO_SETTINGS_MODULE=test_project.settings
+  '';
```
Move this below `nativeCheckInputs`.

mweinelt commented on pkgs/development/python-modules/django-lasuite/default.nix:
```diff
@@ -0,0 +1,83 @@
+    export DJANGO_SETTINGS_MODULE=test_project.settings
+  '';
+
+  pythonRelaxDeps = true;
```
And this below `dependencies`.

mweinelt commented on pkgs/development/python-modules/django-lasuite/default.nix:
```diff
@@ -0,0 +1,83 @@
+    requests
+  ];
+
+  optional-dependencies = {
+    build = [
+      setuptools
+      wheel
+    ];
+    dev = [
+      factory-boy
+      pytest
+      pytest-django
+      responses
+      ruff
+    ];
+  };
```
```suggestion
  optional-dependencies = {
    dev = [
      factory-boy
      pytest
      pytest-django
      responses
      ruff
    ];
  };
```

- build is probably nothing that any other package would reuse and it is outdated as well
- dev dependencies can just selectively be moved into `nativeCheckInputs` instead
- pytest is redundant with pytestCheckHook
- I doubt that ruff is needed for running the tests

mweinelt commented:
Please update to 0.0.8 so we get Python 3.13 support.

mweinelt commented on pkgs/development/python-modules/django-lasuite/default.nix:
```diff
@@ -0,0 +1,69 @@
+  pyjwt,
+  requests,
+  factory-boy,
+  pytest,
```
unused

mweinelt committed:
python313Packages.django_5_2: init at 5.2.1

https://docs.djangoproject.com/en/5.2/releases/5.2/
https://docs.djangoproject.com/en/5.2/releases/5.2.1/

mweinelt commented:
Unfortunately that change will take a few more months to land. The ecosystem just isn't in ready.

If this package requires django>=5, then we can set it `meta.broken = lib.versionOlder django.version "5"`

mweinelt commented on pkgs/development/python-modules/django-lasuite/default.nix:
```diff
@@ -0,0 +1,71 @@
+    changelog = "https://github.com/suitenumerique/django-lasuite/blob/${src.tag}/CHANGELOG.md";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ soyouzpanda ];
+    broken = lib.versionOlder django.version "5";
```
```suggestion
    broken = lib.versionOlder django.version "5.2";
```

Or else we'll run into trouble with the `InMemoryFileNode`.

schoettl commented:
Thanks for your effort @soyouzpanda!! I'm really looking forward for Docs on NixOS!
